pandas/tests/indexing/test_loc.py

```diff
@@ -1130,3 +1130,16 @@ def test_loc_with_period_index_indexer():
     tm.assert_frame_equal(df, df.loc[list(idx)])
     tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
     tm.assert_frame_equal(df, df.loc[list(idx)])
+
+def test_loc_replace_subset_with_another():
+    # GH#36424 Should raise a warning
+    df1 = pd.DataFrame(
+        data = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+        columns = ['A', 'B', 'C'],
+        index = [0, 0, 1]
+    )
+    df2 = df1.copy()
+    df2[:] = np.nan
+    # it fail if a warning is not raised
+    with pytest.warns(Warning):
```

Comment:
we use `tm.assert_produces_warning` instead of `pytest.warns`
 https://github.com/pandas-dev/pandas/blob/d38dc0652fd97b2ea56cff6696261bd628eb2f96/pandas/_testing.py#L2539

Reply:
Done